### DISCUSSION TITLE
Method array_merge called with named unpacked array

### DISCUSSION THREAD

**Discussion by gruni1992** [vimeo/psalm#8954] [Q&A] [answered]
Hey there, I am a bit stuck and would appreciate some help: I am trying to call array_merge with a spread operator on a nested array. Since updating psalm from 4.7.2 to 5.4.0 I get a NamedArgumentNotAllowed error. I am not sure why this is happening, since having string keys in array_merge is allowed since php 8.1 (which is what we're using).

also calling array_values on the spread first, doesn't work in my case since I actually need to keep the keys.

Here is an example, of what I am trying to achieve.
https://psalm.dev/r/8e412ac6b9

**Comment by weirdan** [comment, marked as answer]
> having string keys in array_merge is allowed since php 8.1 

It doesn't seem to be the case: https://3v4l.org/VZsMp

Or am I missing something?

**Reply by gruni1992** [reply]
In your example you are just directly unpacking an array, so there aren't any arrays left to be merged. In my example I have arrays within an array though, which works fine:

https://3v4l.org/kEdN0

However, admittedly, I cannot do the following:

https://3v4l.org/d2Wos

So I would expect to not have an error, if I change the type to `list<array<array-key, mixed>>` or even `array<int, array<array-key, mixed>>`

**Reply by weirdan** [reply]
And the error goes away if you do that: https://psalm.dev/r/5c5a541678

Note that the `@return` docblock tag is spelt without `s` at the end (I assume this was a typo).

**Reply by gruni1992** [reply]
Hey happy new year and sorry for the late reply. After initially trying in my IDE I was then only trying on psalm.dev and because of the typo you pointed out, nothing changed. Now I got it working. Thank you very much for your help! :)
